plugins/core/IO/qLASIO/src/LasIOFilter.cpp

```diff
@@ -598,6 +599,50 @@ CC_FILE_ERROR LasIOFilter::saveToFile(ccHObject* entity, const QString& filename
 		params.lasOffset = lasOffset;
 	}
 
+	// In case of command line call, add automatically all remaining scalar fiels as extra scalar fields
+	if (!parameters.alwaysDisplaySaveDialog)
+	{
+		uint sfCount = pointCloud->getNumberOfScalarFields();
+		for (uint index = 0; index < sfCount; index++)
+		{
+			ccScalarField *sf = static_cast<ccScalarField*>(pointCloud->getScalarField(index));
+			const char* sfName = sf->getName();
+			bool found = false;
+			for (auto& el : params.standardFields)
+				if (strcmp(sfName, el.name()) == 0)
+					found = true;
+			for (auto& el : params.extraFields)
```

Comment:
if (!found)
 ...

Reply:
I do not understand the comment, can you explain what you mean?

Reply:
I think he meant to do

```C++
for (auto& el : params.standardFields)
    if (strcmp(sfName, el.name()) == 0)
        found = true;


if (!found) {
   for (auto& el : params.extraFields)
     // ...
}
```

Reply:
OK, got it now!